### ISSUE TITLE
"ReferenceError: jsPDF is not defined" with Meteor

### ISSUE THREAD

**Issue by chip** [someatoms/jsPDF-AutoTable#13]
Hi - Really nice work on the docs for this repo. :+1:

Do you know if anyone has gotten this to work within a Meteor project?

I'm currently getting **"ReferenceError: jsPDF is not defined"**

```
=> Modified -- restarting.
W20150407-11:48:14.168(-5)? (STDERR)          
W20150407-11:48:14.169(-5)? (STDERR) /Users/chip/.meteor/packages/meteor-tool/.1.0.45.1bea6ne++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:245
W20150407-11:48:14.169(-5)? (STDERR)                                            throw(ex);
W20150407-11:48:14.169(-5)? (STDERR)                                                  ^
W20150407-11:48:14.172(-5)? (STDERR) ReferenceError: jsPDF is not defined
W20150407-11:48:14.172(-5)? (STDERR)     at app/lib/jspdf.plugin.autotable.js:287:4
W20150407-11:48:14.172(-5)? (STDERR)     at app/lib/jspdf.plugin.autotable.js:289:3
W20150407-11:48:14.172(-5)? (STDERR)     at /Users/chip/code/invoicethat.com/.meteor/local/build/programs/server/boot.js:222:10
W20150407-11:48:14.172(-5)? (STDERR)     at Array.forEach (native)
W20150407-11:48:14.172(-5)? (STDERR)     at Function._.each._.forEach (/Users/chip/.meteor/packages/meteor-tool/.1.0.45.1bea6ne++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/underscore/underscore.js:79:11)
W20150407-11:48:14.172(-5)? (STDERR)     at /Users/chip/code/invoicethat.com/.meteor/local/build/programs/server/boot.js:117:5
=> Exited with code: 8
```

It appears that it is failing on line 284 of this plugin due to `doc` being undefined:

```javascript
283     function getStringWidth(txt) {
284         return doc.getStringUnitWidth(txt) * doc.internal.getFontSize();                       
285     }
```

Here's the code I'm using as an example:

**client/lib/PDF.coffee**
```coffee
class PDF
  columns: ->
    [
      {
        title: 'ID'
        key: 'id'
      }
      {
        title: 'Name'
        key: 'name'
      }
      {
        title: 'Country'
        key: 'country'
      }
      {
        title: 'IP-address'
        key: 'ip_address'
      }
      {
        title: 'Email'
        key: 'email'
      }
    ]

  data: ->
    [
      {
        'id': 1
        'name': 'Shaw'
        'country': 'Tanzania'
        'ip_address': '92.44.246.31'
        'email': 'abrown@avamba.info'
      }
      {
        'id': 2
        'name': 'Nelson'
        'country': 'Kazakhstan'
        'ip_address': '112.238.42.121'
        'email': 'jjordan@agivu.com'
      }
    ]

  generate: ->
    doc = new jsPDF('p', 'pt')
    doc.autoTable @columns(), @data()
    uri = doc.output 'datauristring'
    console.log "uri ", uri

module?.PDF = PDF
window?.PDF = PDF
```

When I comment out the `doc.autoTable` it calls `doc.output` successfully, so it appears that **jsPDF** is loaded properly.

Any suggestions would be helpful. Thank you.

**Comment by someatoms** [comment]
Sounds strange. I just tried to generate a basic table with meteor and it worked. Could you try to port your example to javascript and see if that works. I haven't had the opportunity to use coffee script yet...

**Comment by someatoms** [comment]
I tried converting your example to javascript now and including it in the meteor hello world example now which worked. This means that it must be some kind of loading order issue in your case? This is the code that worked:

    if (Meteor.isClient) {

      Template.hello.events({
        'click button': function () {
          var columns = [
            {
              title: 'ID',
              key: 'id'
            }, {
              title: 'Name',
              key: 'name'
            }, {
              title: 'Country',
              key: 'country'
            }, {
              title: 'IP-address',
              key: 'ip_address'
            }, {
              title: 'Email',
              key: 'email'
            }
          ];

          var data = [
            {
              'id': 1,
              'name': 'Shaw',
              'country': 'Tanzania',
              'ip_address': '92.44.246.31',
              'email': 'abrown@avamba.info'
            }, {
              'id': 2,
              'name': 'Nelson',
              'country': 'Kazakhstan',
              'ip_address': '112.238.42.121',
              'email': 'jjordan@agivu.com'
            }
          ];

          var doc, uri;
          doc = new jsPDF('p', 'pt');
          doc.autoTable(columns, data);
          uri = doc.output('datauristring');
          console.log("uri ", uri);
          
        }
      });
    }

**Comment by chip** [comment]
Thanks for your prompt response and it was a really good idea to start over with a clean slate. It allowed me to step back a bit and think about how I was loading the files. I tried your suggestion and was able to get it to work by putting those files under `client/compatibility/`, so it was my mistake all along.

At any rate, I have created a [jspdf package on atmosphere](https://atmospherejs.com/chipcastledotcom/meteor-jspdf) and was considering doing the same for your plugin if I have the time and you're ok with it.

Thanks again for your help. Really nice work on this plugin!  :smiley:
